File: src/controllers/yearKpi.js
var { prepareConnection } = require("../../database/conection");

const controlador = {
  listar: {},
};

controlador.kpiYear = (req, res) => {

  prepareConnection(9)
    .then((resolve) => {
      res.status(200).json({
        status: 200,
        message: "Ok",
        response: resolve,
      });
      console.log(resolve);
    })
    .catch((reject) => {
      res.status(500).json({
        status: 500,
        message: "Internal Server Error",
        response: reject,
      });
      console.log(reject);
    });
};

module.exports = controlador;

File: src/controllers/yearKpi.py
from flask import jsonify
from database.conection import prepareConnection

controlador = {
    'listar': {},
}

def kpi_year(req, res):
    try:
        result = prepareConnection(9)
        res.status_code = 200
        return jsonify({
            'status': 200,
            'message': 'Ok',
            'response': result
        })
    except Exception as e:
        res.status_code = 500
        return jsonify({
            'status': 500,
            'message': 'Internal Server Error',
            'response': str(e)
        })

# Asignación de funciones al controlador
controlador['kpiYear'] = kpi_year

# Exportación del controlador
